Fix compatibility with bevy_rapier2d. #7

jamessizeland opened the pull request:
change ordering of PostUpdate propagating transforms to not conflict with bevy_rapier2d by using the chain() method instead.

Have tested this in my project: https://github.com/jamessizeland/bevy-chase-me-game

jamessizeland commented on src/lib.rs:
```diff
@@ -54,10 +54,7 @@ impl Plugin for TraumaPlugin {
-                    .before(propagate_transforms)
-                    .before(sync_simple_transforms),
-            );
+                (shake, propagate_transforms, sync_simple_transforms).chain());
```
Hmm yeah you're right, I'll look at this again. This is definitely the line that causes the error with rapier2d though.

jamessizeland replied:
I might still be missing something, (I'm still new to Bevy) but looking at the way Bevy Transform works, sync_simple_transforms is part of a set TransformSystem::TransformPropagate https://docs.rs/bevy_transform/0.14.1/src/bevy_transform/plugins.rs.html#12 which might give the desired bahaviour of guaranteeing that shake happens before these systems fire in PostUpdate.

johanhelsing commented:
I'm not sure i understand exactly why this was causing problems with rapier? What was wrong with the old code?

jamessizeland commented:
This error: 
```bash
thread 'main' panicked at C:\Users\james.cargo\registry\src\index.crates.io-6f17d22bba15001f\bevy_ecs-0.14.1\src\schedule\schedule.rs:383:33: Error when initializing schedule PostUpdate: Tried to order against SystemTypeSet(fn FunctionSystem<fn(ParamSet<(Query<(Transform, &mut GlobalTransform), (Or<(Changed<Transform>, Added<GlobalTransform>)>, Without<Parent>, Without<Children>)>, Query<(Ref<Transform>, &mut GlobalTransform), (Without<Parent>, Without<Children>)>)>, RemovedComponents<Parent>), sync_simple_transforms>()) in a schedule that has more than one SystemTypeSet(fn FunctionSystem<fn(ParamSet<(Query<(Transform, &mut GlobalTransform), (Or<(Changed<Transform>, Added<GlobalTransform>)>, Without<Parent>, Without<Children>)>, Query<(Ref<Transform>, &mut GlobalTransform), (Without<Parent>, Without<Children>)>)>, RemovedComponents<Parent>), sync_simple_transforms>()) instance. SystemTypeSet(fn FunctionSystem<fn(ParamSet<(Query<(Transform, &mut GlobalTransform), (Or<(Changed<Transform>, Added<GlobalTransform>)>, Without<Parent>, Without<Children>)>, Query<(Ref<Transform>, &mut GlobalTransform), (Without<Parent>, Without<Children>)>)>, RemovedComponents<Parent>), sync_simple_transforms>()) is a SystemTypeSet and cannot be used for ordering if ambiguous. Use a different set without this restriction.
```
occurs when bevy_rapier2d is included in the project.  I'm seeing the same issue as mentioned in https://github.com/johanhelsing/bevy_trauma_shake/issues/7. I'm not 100% sure why this is happening but from my digging I think its related to the way the bevy_transform plugin's PostUpdate is configured.  This fix works for me so I thought I'd suggest it back.
